lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees, :registered_vehicles, :registration_date
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_hash)
+    @name = facility_hash[:name]
+    @address = facility_hash[:address]
+    @phone = facility_hash[:phone]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") 
+      @registered_vehicles << vehicle 
+      vehicle.registration_date = Date.today
+      collect_fees(vehicle)
+      @registered_vehicles
+    end
+  end 
+
+  def collect_fees(vehicle)
+    if vehicle.plate_type == :antique
+      @collected_fees += 25
+    elsif vehicle.plate_type == :ev
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end 
+  end 
+
+  def administer_written_test(registrant)
+    if @services.include?("Written Test") && registrant.permit? && registrant.age >= 16
+      registrant.license_data[:written] = true
+    else
+      false
+    end
+  end
+
+  def administer_road_test(registrant)
+    if registrant.license_data[:written] == true
```

Comment:
This should be looking for more conditions to determine if they can take the road test. The facility needs to have this service for one.